script/testing/util/common.py

```diff
@@ -22,55 +26,83 @@ def run_command(command,
+            proc.parent().send_signal(signal.SIGCHLD)
+
+
+def get_pids_on_port(port):
     """Get the list of PIDs (if any) listening on the target port"""
-    
-    # Copied from https://gist.github.com/jossef/593ade757881bb7ddfe0
-    # I would like to use psutil to make this more portable but that would require
-    # us to install an additional package with pip
-    
-    command = "lsof -i :%s | awk '{print $2}'" % port
-    output = subprocess.check_output(command, shell=True).strip()
-    if output:
-        output = re.sub(' +', ' ', output.decode('utf-8'))
-        for pid in output.split('\n'):
-            try:
-                yield int(pid)
-            except:
-                pass
-
-
-def check_pid(pid):
-    """Check whether pid exists in the current process table."""
-    
-    # Copied from psutil
-    # https://github.com/giampaolo/psutil/blob/5ba055a8e514698058589d3b615d408767a6e330/psutil/_psposix.py#L28-L53
-    
-    if pid == 0:
-        return True
-    try:
-        os.kill(pid, 0)
-    except OSError as err:
-        if err.errno == errno.ESRCH:
-            # ESRCH == No such process
-            return False
-        elif err.errno == errno.EPERM:
-            # EPERM clearly means there's a process to deny access to
-            return True
-        else:
-            # According to "man 2 kill" possible error values are
-            # (EINVAL, EPERM, ESRCH) therefore we should never get
-            # here. If we do let's be explicit in considering this
-            # an error.
-            raise err
-    else:
-        return True
+
+    pids = []
+    for proc in psutil.process_iter():
+        try:
+            for conns in proc.connections(kind="inet"):
+                if conns.laddr.port == port:
+                    pids.append(proc.pid)
+        except psutil.ZombieProcess:
+            # ignore the zombie process
+            continue
+    return pids
+
+
+def check_pid_exists(pid):
+    """ Checks to see if the pid exists """
+    return psutil.pid_exists(pid)
+
+
+def run_check_pid_exists(pid):
+    cmd = "sudo python3 {SCRIPT} {PID}".format(SCRIPT=FILE_CHECK_PID_EXISTS,
+                                               PID=pid)
+    rc, stdout, _ = run_command(cmd,
+                                error_msg="",
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                cwd=None,
+                                printable=False)
+    if rc != ErrorCode.SUCCESS:
+        LOG.error(
+            "Error occured in run_check_pid_exists for [PID={}]".format(pid))
+        return False
+    res_str = stdout.readline().decode("utf-8").rstrip("\n")
+    return res_str == CommandLineStr.TRUE
+
+
+def run_kill_pids_on_port(port):
+    cmd = "sudo python3 {SCRIPT} {PORT}".format(
+        SCRIPT=FILE_KILL_SERVER_ON_PORT, PORT=port)
+    rc, _, _ = run_command(cmd)
```

Comment:
Why not check the return code here like you do for `run_check_pid_exists`

Reply:
The difference between this one and the `run_check_pid_exists` is that we will use the return value of the `run_check_pid_exists`, but not this function. Nevertheless, we can verify the `rc` here. In theory, it should always be 0; otherwise, it means reaping lingering processes fails, which can be an exception.